src/cdn_definitions/schema.yaml

```diff
@@ -11,7 +11,211 @@ $id: http://release-engineering.github.io/cdn-definitions/schema.json
 
 
 definitions:
+  arch:
+    type: string
+    pattern: "^(x86|x86\\_64|ia64|i386|ppc|ppc64|ppc64le|s390|s390x|aarch64)$"
+
+  arch_list:
+    type: array
+    items:
+      $ref: "#/definitions/arch"
+
+  release_version:
+    type: string
+    pattern: "^[0-9]+\\.[0-9]+$"
+
+  major_version:
+    type: string
+    pattern: "^[0-9]+$"
+
+  signing_key:
+    type: string
+    pattern: "^(37036783|A5787476|FD431D51|F21541EB|37017186|897DA07A|DB42A60E|\\[\\])$"
```

Comment:
I think this one doesn't belong in the schema.  Could we just enforce the format instead and not make this an enum of values?

I don't like the idea that adding a new signing key requires updating the schema, or that one is unable to use cdn-definitions in their own test environment with their own personal signing key.

By the way, what's the meaning of ```[]``` as signing_key ?

Reply:
`[]` as a signing key is from `cdsutils.ProductContentSet.signatures`:
```
if self.data.platform == 'middleware':
    return []  # no RPM signing of zips
```

Reply:
Right, I thought `[]` might be something like that, but I don't think this is correct for reasons:

1. The underlying data (JSON or YAML) supports lists natively, so if we want to have an empty list as a value, we should just write an empty list into the file... not a list encoded into a string.  This is a case of "JSON within JSON" or "YAML within YAML" for no clear reason.
2. Is this field meant to represent a single signing key or a list of them?  It seems that it should be one or the other.  If a single signing key, then either null or an empty string is better to represent 'no signing key' than `[]`.